benchmark/simple_with_type_coercer.rb

```diff
@@ -0,0 +1,22 @@
+$LOAD_PATH.unshift(File.join(File.dirname(__FILE__), '..', 'lib'))
```

Comment:
Don't know what's the policy with adding these benchmarks, this can be removed if seen obsolete.

The results between not cached and cached version were about these:
```
No cache                   1.457k (± 8.8%) i/s -      7.250k in   5.017311s
Caching enabled            3.440k (± 5.6%) i/s -     17.435k in   5.085603s
```